client/src/components/_common/AppIcon/AppIcon.jsx

```diff
@@ -4,10 +4,14 @@ import PropTypes from 'prop-types';
 import Icon from '_common/Icon';
 import './AppIcon.scss';
 
-const AppIcon = ({ appId }) => {
+const AppIcon = ({ appId, category }) => {
   const appIcons = useSelector((state) => state.apps.appIcons);
   const findAppIcon = (id) => {
-    let appIcon = 'applications';
+    if (!category) {
+      console.error('Category is undefined for appId:', id);
+      return 'applications';
```

Comment:
This might not be picking up the default icon, steps to reproduce:

1. Run `./manage.py import-apps -c` inside the `core_portal_django` container to pick up an updated array of apps
2. Select the "Interactive Analysis" category
![Screenshot 2023-10-04 at 6 41 32 PM](https://github.com/TACC/Core-Portal/assets/20326896/3f597d51-86e6-432b-aa70-3198e5ffd9c5)

Reply:
So the way I coded it to work with category icons is: it's taking the name of the category the application is in and converting it into a className. Basically the same way y'all did it in the AppBrowser file before with className for the Icon component in that file. So since we don't have a custom icon in the icon.font.css file with the style rule '.icon-interactive-analysis' it's going to return the className as such for those three applications in the screenshot (thus making them blank). I hadn't seen Interactive Analysis as a category in any of the portals I looked at, so I wasn't aware that it would be an issue, but I see now if we add a new category without it having its own icon associated with that category, the applications without their own unique icon will be blank.

So we can either 1) add a css style in icon.css called '.icon-interactive-analysis' and it will fix the issue of those three icons being blank (this is a quick fix) or 2) I can try and figure out a way for it to search the style rules in icon.fonts.css and/or icon.css and if it doesn't find a style it should default to applications? (Might be complex, not sure).

Reply:
Gotcha, yeah I believe the latter would be a better dynamic solution to aim for. It will be a common case for a category to have no associated icon, and the apps within that category to have no icon as well

Reply:
<img width="1423" alt="Screenshot 2023-10-06 at 9 13 44 AM" src="https://github.com/TACC/Core-Portal/assets/50084480/4cc7152d-3e00-4847-9ca1-ad48cfceac18">